core/src/main/java/org/apache/calcite/rel/type/RelDataTypeSystem.java

```diff
@@ -62,6 +64,14 @@ public interface RelDataTypeSystem {
   /** Returns the maximum precision of a NUMERIC or DECIMAL type. */
   int getMaxNumericPrecision();
 
+  /** Returns the minimum scale of a NUMERIC or DECIMAL type. */
+  int getMinNumericScale();
```

Comment:
You should say that the default type system returns 0,
and that there are no 'special values' (like SCALE_NOT_SPECIFIED).

Reply:
RelDataTypeSystem is an interface It's strange to add a Java document here. So I added it in RelDataTypeSystemImpl. We have a SCALE_NOT_SPECIFIED in RelDataType for the types where the scale is not allowed.